mmv1/products/discoveryengine/DataStore.yaml

```diff
@@ -140,6 +147,92 @@ properties:
       - :PUBLIC_WEBSITE
     immutable: true
     required: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'documentProcessingConfig'
+    immutable: true
+    description: |
+      Configuration for Document understanding and enrichment.
+    required: false
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The full resource name of the Document Processing Config. Format:
+          `projects/{project}/locations/{location}/collections/{collection_id}/dataStores/{data_store_id}/documentProcessingConfig`.
+        output: true
+        required: false
+
+      - !ruby/object:Api::Type::NestedObject
+        name: 'defaultParsingConfig'
+        description: |
+          Configurations for default Document parser. If not specified, we will
+          configure it as default DigitalParsingConfig, and the default parsing
+          config will be applied to all file types for Document parsing.
+        required: false
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'digitalParsingConfig'
+            allow_empty_object: true
+            send_empty_value: true
+            exactly_one_of:
+              - default_parsing_config.0.digital_parsing_config
+              - default_parsing_config.0.ocr_parsing_config
+            description: |
+              Configurations applied to digital parser.
+            required: false
+            properties: []
+          - !ruby/object:Api::Type::NestedObject
+            name: 'ocrParsingConfig'
+            exactly_one_of:
+              - default_parsing_config.0.digital_parsing_config
+              - default_parsing_config.0.ocr_parsing_config
+            description: |
+              Configurations applied to OCR parser. Currently it only applies to PDFs.
+            required: false
+            properties:
+              - !ruby/object:Api::Type::Boolean
+                name: 'useNativeText'
+                required: false
+                description: |
+                  If true, will use native text instead of OCR text on pages containing native text.
+
+      - !ruby/object:Api::Type::Map
+        name: 'parsingConfigOverrides'
+        description: |
```

Comment:
The description mentions some constraints that cannot be modeled in a map (a NestedObject could model those, but then it wouldn't be consistent with the docs)

Also, the description in [the docs](https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.collections.dataStores#DataStore.DocumentProcessingConfig) mentions Layout parsing, which is only available in alpha